File: solutions/Question 206/Solution1.cpp
/*
 * Problem: 206. Reverse Linked List
 * Difficulty: Easy
 * Topics: Linked List, Recursion
 * URL: https://leetcode.com/problems/reverse-linked-list/
 * Runtime: 0 ms
 * Memory: 13.4 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* first = nullptr;
        ListNode* last = head;
        while (last != nullptr) {
            ListNode* temp = last->next; 
            last->next = first;
            first = last;
            last = temp;
        }
        return first;
    }
};

File: solutions/Question 58/Solution1.py
# Problem: 58. Length of Last Word
# Difficulty: Easy
# Topics: String
# URL: https://leetcode.com/problems/length-of-last-word/
# Runtime: 0 ms
# Memory: 18 MB
# Solved via CodePath Auto-Committer

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        a = s.split()
        return len(a[-1])